pkg/kubelet/apis/kubeletconfig/v1alpha1/defaults.go

```diff
@@ -205,9 +205,25 @@ func SetDefaults_KubeletConfiguration(obj *KubeletConfiguration) {
 	if string(obj.HairpinMode) == "" {
 		obj.HairpinMode = PromiscuousBridge
 	}
+	if obj.NodeLabels == nil {
+		obj.NodeLabels = make(map[string]string)
+	}
 	if obj.EvictionHard == nil {
-		temp := "memory.available<100Mi,nodefs.available<10%,nodefs.inodesFree<5%,imagefs.available<15%"
-		obj.EvictionHard = &temp
+		obj.EvictionHard = map[string]string{
+			"memory.available":  "100Mi",
+			"nodefs.available":  "10%",
+			"nodefs.inodesFree": "5%",
+			"imagefs.available": "15%",
+		}
+	}
+	if obj.EvictionSoft == nil {
+		obj.EvictionSoft = make(map[string]string)
+	}
+	if obj.EvictionSoftGracePeriod == nil {
+		obj.EvictionSoftGracePeriod = make(map[string]string)
```

Comment:
We don't typically default nils to empty collections… those should be semantically equivalent

Reply:
Flag parsing shims need to target non-nil maps as written. We could, instead, make the shims operate on pointers to maps, and allocate the maps if necessary.

Reply:
Let me know what you think about that, or just leaving as-is.

Reply:
pointers would be better than assuming/forcing non-nil values

Reply:
Changed to pointers, and also removed the unnecessary defaulters.